stage/base_stage.py

```diff
@@ -0,0 +1,31 @@
+"""The abstract base class for stages in fuzzing pipeline."""
+import argparse
+from abc import ABC, abstractmethod
+
+from agent.base_agent import BaseAgent
+from result_classes import Result
+
+
+class BaseStage(ABC):
+  """The abstract base class for stages in fuzzing pipeline."""
+
+  def __init__(self, args: argparse.Namespace) -> None:
+    self.args = args
+    self.agents: list[BaseAgent] = []
+
+  def add_agent(self, agent: BaseAgent) -> 'BaseStage':
```

Comment:
do we need the ' around the `-> 'BaseStage'` here ?

Reply:
Yes, because this is in `BaseStage`. Otherwise linter will complain:
![image](https://github.com/user-attachments/assets/9cf1a3dc-6e6e-412e-85a5-4b2c3a75ccc6)